application/comit_node/src/http_api/rfc003/swap.rs

```diff
@@ -270,94 +319,137 @@ enum SwapCommunicationState {
-//             communication: item.swap_communication.into(),
-//             alpha_ledger: item.alpha_ledger_state.into(),
-//             beta_ledger: item.beta_ledger_state.into(),
-//         }
-//     }
-// }
-
-// impl<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>
-//     From<rfc003::SwapRequestParameters<AL, BL, AA, BA>> for SwapDescription
-// {
-//     fn from(item: rfc003::SwapRequestParameters<AL, BL, AA, BA>) -> Self {
-//         Self {
-//             alpha_ledger: item.alpha_ledger.to_http_ledger().unwrap(),
-//             beta_ledger: item.beta_ledger.to_http_ledger().unwrap(),
-//             alpha_asset: item.alpha_asset.to_http_asset().unwrap(),
-//             beta_asset: item.beta_asset.to_http_asset().unwrap(),
-//         }
-//     }
-// }
-
-// impl<AL: Ledger, BL: Ledger> From<rfc003::SwapCommunication<AL, BL>> for
-// SwapCommunication {     fn from(item: rfc003::SwapCommunication<AL, BL>) ->
-// Self {         let rfc003::SwapCommunication {
-//             alice_identities,
-//             bob_identities,
-//             alpha_expiry,
-//             beta_expiry,
-//             was_rejected,
-//         } = item;
-
-//         let current_state = if was_rejected {
-//             SwapCommunicationState::Rejected
-//         } else {
-//             match bob_identities {
-//                 Some(_) => SwapCommunicationState::Accepted,
-//                 None => SwapCommunicationState::Sent,
-//             }
-//         };
-
-//         Self {
-//             current_state,
-//             alpha_redeem_identity: bob_identities.clone().map(|x| json!(x)),
-//             beta_redeem_identity: json!(alice_identities.redeem),
-//             alpha_refund_identity: json!(alice_identities.refund),
-//             beta_refund_identity: bob_identities.map(|x| json!(x)),
-//             alpha_expiry,
-//             beta_expiry,
-//         }
-//     }
-// }
-
-// impl<L: Ledger> From<rfc003::LedgerState<L>> for LedgerState {
-//     fn from(item: rfc003::LedgerState<L>) -> Self {
-//         let rfc003::LedgerState {
-//             current_state,
-//             htlc_location,
-//             fund_transaction,
-//             redeem_transaction,
-//             refund_transaction,
-//         } = item;
-
-//         Self {
-//             current_state,
-//             htlc_location: htlc_location.map(|htlc_location| format!("{:?}",
-// htlc_location)),             fund_tx: fund_transaction.map(|fund_tx|
-// format!("{:?}", fund_tx)),             redeem_tx:
-//  redeem_transaction.map(|redeem_tx| format!("{:?}", redeem_tx)),
-// refund_tx: refund_transaction.map(|refund_tx| format!("{:?}", refund_tx)),
-//         }
-//     }
-// }
+impl<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> From<rfc003::messages::Request<AL, BL, AA, BA>>
+    for SwapDescription
+{
+    fn from(request: rfc003::messages::Request<AL, BL, AA, BA>) -> Self {
+        Self {
+            alpha_ledger: request.alpha_ledger.to_http_ledger().unwrap(),
```

Comment:
:thinking: Should we really unwrap here?

Reply:
Clippy doesn't let you anyway, so I replaced it with a `new()`.